### ISSUE TITLE
Button wont get disabled at startup

### ISSUE THREAD

**Issue by ShariatPanah** [dotnet/maui#20347]
### Description

i'm trying to disable a button based on an Entry's Text.Length but it won't get disabled at startup when the entry is empty, but then when i type in the entry and clear it out, the button gets disabled, i tried using DataTriggers and also binding the Button.IsEnabled to Text.Length using a converter but none of them work at startup

### Steps to Reproduce

either use this:
```
<Button
    Command="{Binding AddSectionCommand}"
    CommandParameter="{Binding Section}"
    IsEnabled="{Binding Source={x:Reference txtTitle}, Path=Text.Length, Converter={StaticResource lengthConverter}}"
    Text="Save"/>
```

or this:
```
<Button
    Command="{Binding AddSectionCommand}"
    CommandParameter="{Binding Section}"
    Text="Save">
    <Button.Triggers>
        <DataTrigger
            Binding="{Binding Source={x:Reference txtTitle}, Path=Text.Length}"
            TargetType="Button"
            Value="0">
            <Setter Property="IsEnabled" Value="False" />
        </DataTrigger>
    </Button.Triggers>
</Button>
```

### Version with bug

8.0.6

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Android 12.0

**Comment by ShariatPanah** [comment]
And as i mentioned before, it doesn't matter if you bind the IsEnabled property to the text.length using a converter or you use DataTrigger, it won't disable the button at startup when the entry is empty

**Comment by jenscski** [comment]
Not sure if this can be considered a bug or not, but the issues is that Entry.Text is default NULL, so the trigger that checks for Text.Length=0 will not succeed, because Text is NULL.

If you explicit set the Entry.Text to an empty string, then it works as excepted.

`<Entry x:Name="txtTitle" Text="" />`

**Comment by ShariatPanah** [comment]
@jenscski you are right, but as i said even with a converter it won't work, the converter won't even get called at startup, I'm binding the entry to a property of an object, at first this string property is null, but doesn't it have to call the converter at any time? like to evaluate and render the View? i don't know maybe this behavior is by design. please let me know

**Comment by jenscski** [comment]
What is the lengthConverter ?

Have you looked at .NET MAUI Community Toolkit, and their converters?
E.g. https://learn.microsoft.com/en-us/dotnet/communitytoolkit/maui/converters/is-string-not-null-or-empty-converter

**Comment by ShariatPanah** [comment]
the lengthConverter is a resource i added to ContentPage.Resources and gave it this name as x:Key, it's of my TextLengthToBoolConverter Class type, here is the code:
'''
public class TextLengthToBoolConverter : IValueConverter
{
    public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
    {
        if (value == null)
            return false;

        if ((int)value == 0)
            return false;

        return true;
    }

    public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
    {
        throw new NotImplementedException();
    }
}
'''

i can use this converter to check if the entry text property is null or empty or is filled. and based on this the button's IsEnabled property is being set. it does work after i type some character into the entry and clear it, but at startup this converter won't even get called,

**Comment by ShariatPanah** [comment]
i updated my repo and added converter to reproduce the issue, here is the link: [https://github.com/ShariatPanah/ButtonDisableBug](https://github.com/ShariatPanah/ButtonDisableBug)

**Comment by ShariatPanah** [comment]
> When you are using a backing field to bind to, you should also use this field to check for changes.

I'm sorry i didn't get your point.
i am binding IsEnabled to Entry's Text yes, i did everything right please take a look at my repo

**Comment by jenscski** [comment]
I created a pull request for you, that show how to do this using MVVM and converters.

https://github.com/ShariatPanah/ButtonDisableBug/pull/1

**Comment by ShariatPanah** [comment]
> I created a pull request for you, that show how to do this using MVVM and converters.

I'll give it a try and will let you know the result, thanks.
